### PR TITLE
Feature/rc lights more names and status values

### LINKED ISSUE #2172: [SDL 0165] Remote Control – Lights – More Names and Status Values
Proposal: [Remote Control – Lights – More Names and Status Values](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0165-rc-lights-more-names-and-status-values.md)

> The light status signal in the vehicle can provide more values than just ON / OFF for certain lights. Some vehicles can provide the status of more lights than currently defined light names. This proposal adds some new light status values and some new light names to allow an application to get the status of additional lights.

Review: smartdevicelink/sdl_evolution#479

Steering Committee Decision:
> The Steering Committee voted to accept this proposal with the revisions.  The revisions will include the error and documentation as described in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/479#issuecomment-384992052) and [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/479#issuecomment-385676952), respectively.

> The proposal .md was updated on May 4, 2018 to include the agreed upon revisions.

### PR DESCRIPTION
**Pull request by KVGrygoriev**
Fixes #2172

This PR is **not ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
- [x] ATF tests
- [ ] unit tests
- [ ] HMI

### Summary
This PR contains changes related to extending lights naming in the API.
Proposal : https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0165-rc-lights-more-names-and-status-values.md

### Tasks Remaining:
- [ ] Unit tests

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc [hunk 1]
```diff
@@ -515,11 +520,22 @@ void SetInteriorVehicleDataRequest::CutOffReadOnlyParams(
...
       if (enums_value::kClimate == module_type) {
         module_data[message_params::kClimateControlData].erase(it);
-        LOG4CXX_DEBUG(logger_, "Cutting-off READ ONLY parameter: " << it);
       } else if (enums_value::kRadio == module_type) {
         module_data[message_params::kRadioControlData].erase(it);
-        LOG4CXX_DEBUG(logger_, "Cutting-off READ ONLY parameter: " << it);
+      } else if (enums_value::kLight == module_type) {
+        const uint64_t light_status = module_data[message_params::kLightControlData][message_params::kStatus].asUInt();
+        
+        if ((mobile_apis::LightStatus::RAMP_UP == light_status) ||
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @KVGrygoriev Please use `utils::helpers::Compare` instead

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc [hunk 2]
```diff
@@ -515,11 +520,22 @@ void SetInteriorVehicleDataRequest::CutOffReadOnlyParams(
     if (module_type_params.keyExists(it)) {
       if (enums_value::kClimate == module_type) {
         module_data[message_params::kClimateControlData].erase(it);
-        LOG4CXX_DEBUG(logger_, "Cutting-off READ ONLY parameter: " << it);
       } else if (enums_value::kRadio == module_type) {
         module_data[message_params::kRadioControlData].erase(it);
-        LOG4CXX_DEBUG(logger_, "Cutting-off READ ONLY parameter: " << it);
+      } else if (enums_value::kLight == module_type) {
+        const uint64_t light_status = module_data[message_params::kLightControlData][message_params::kStatus].asUInt();
```
[the remark this answers is not in the document]
Comment (KVGrygoriev) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @AByzhynar see e3e5724

[the remark this answers is not in the document]
Comment (KVGrygoriev) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @AByzhynar see e3e5724

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc (hunk 1 above)
Comment (KVGrygoriev) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @AByzhynar see e3e5724

**Pull request merged**